src/gpu/tsvd/tsvd.cu

```diff
@@ -12,6 +12,14 @@
 namespace tsvd
 {
 
+/**
+ * Division utility to get explained variance ratio
+ *
+ * @param XVar
+ * @param XVarSum
+ * @param ExplainedVarRatio
+ * @param context
+ */
```

Comment:
Possible divide by zero in this function?

Reply:
ah, good point. will fix.